ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/Utils.java

```diff
@@ -172,7 +172,8 @@ public static Boolean shouldReplicate(ReplicationSpec replicationSpec, Table tab
       return false;
     }
 
-    if (tableHandle.isNonNative()) {
+    // if its metadata only, then dump metadata of non native tables also.
+    if (tableHandle.isNonNative() && !replicationSpec.isMetadataOnly()) {
```

Comment:
Better to move this check only if it is in replication scope. Or else, will impact export table flow.

Reply:
to keep dump through falcon (export) and dump through beacon (repl dump) ..i think its better to keep the check this way